addons/explosives/functions/fnc_onRespawn.sqf

```diff
@@ -0,0 +1,30 @@
+#include "script_component.hpp"
+/*
+ * Author: Whigital
+ * Clears cached variable containing placeable explosives.
+ *
+ * Arguments:
+ * 0: Unit <OBJECT>
+ * 1: Corspe <OBJECT>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ *
+ * Public: No
+ */
+
+params [
+    "_unit",
+    "_corpse"
+];
+
+TRACE_2("params",_unit,_corpse);
+
+if (_unit == player) then {
+    [
+        {ACE_player == player},
+        {[ACE_player, QGVAR(explosiveActions)] call EFUNC(common,eraseCache);}
+    ] call CBA_fnc_waitUntilAndExecute;
```

Comment:
I can not make sense of this WUAE. Please explain.

Reply:
Placable explosives actions are generated by `ace_explosives_fnc_fnc_addExplosiveActions`, which wraps the code in a cached call to `ace_common_fnc_cachedCall`.
https://github.com/acemod/ACE3/blob/19523f028a2d1316e8a15a618d643b611b543cf7/addons/explosives/functions/fnc_addExplosiveActions.sqf#L42

If the inventory changes, var (`ace_explosives_explosiveActions`) that stores the cached actions gets cleared. All good so far, works as intended.

Somewhere in the process of getting killed and respawning though, the reference to `ACE_player` changes, and the `cba_events_loadoutEvent` that is triggered after respawn does not clear the `ace_explosives_explosiveActions` var.

Don't know if it has to do with the `ACE_player` var referencing the old corpse still, or if the var gets restored. Only way i was able to *fix* it was to manually delay the call that is made when a `cba_events_loadoutEvent` is triggered.
https://github.com/acemod/ACE3/blob/19523f028a2d1316e8a15a618d643b611b543cf7/addons/common/functions/fnc_cachedCall.sqf#L49